README

This is a bundle-like set of prerequisites to install the modules that RJBS
requires for day to day operation.



Changes
Revision history for {{$dist->name}}

{{$NEXT}}

20221231.000 2022-12-31 20:59:30-05:00 America/New_York
            - just update some metadata

20211226.000 2021-12-26 10:02:21-05:00 America/New_York
            - add Module::CPANfile -- I keep forgetting how to get
              cpanfile-dump, which I always want

20210619.000 2021-06-19 08:54:23-04:00 America/New_York
            - update list based on things I keep installing
            - bump lowest version to v5.20.0, which it effectively was

20210316.000 2021-03-16 22:36:57-04:00 America/New_York
            - JSON::MaybeXS, not JSON
            - turn on declared_refs when possible

20200521.000 2020-05-21 09:16:14-04:00 America/New_York
            - drop Crypt::SSLeay -- that'd be nice if it pans out!

20150423.001 2015-04-23 13:20:43-04:00 America/New_York
            - allow rjbs.pm to work on 5.20, with fewer features; this is
              going to be a hard minimum

20150423.000 2015-04-23 09:25:13-04:00 America/New_York
            - add the rjbs.pm module, to quickly turn on all kinds of nonsense

20131116.000 2013-11-16 17:09:04 America/New_York
            - stop trying to install perlbrew
            - require dzil 5!

20131018.001 2013-10-18 11:40:55 America/New_York
            - update bugtracker and repo
            - remove Moose::Autobox

20120926.000 2012-09-26 12:09:52 America/New_York
            - add App::grindperl, App::perlbrew, perl 5.14.0
            - add Pod::Cpandoc

20120310.000 2012-03-10 14:04:11 America/New_York
            - add Unicode::Tussle, because it's cool
            - add MooseX::StrictConstructor; please use it!
            - switch from List::MoreUtils to List::AllUtils

20110621.000 2011-06-21 22:16:40 America/New_York
            - add Module::Which, for which_pm, which I use a lot
            - add App::Uni, for uni, which I also use a lot

20110516.000 2011-05-16 16:53:43 America/New_York
            - require LWP::Protocol::https to make sure I can use LWP on SSL

20110324.000 2011-03-24 11:01:14 America/New_York
            - improve behavior of App::Nopaste by adding its optional prereqs

20101201.000 2010-12-01 18:21:32 America/New_York
            - require perl 5.12.0
            - drop CPAN, CPAN::Reporter, CPAN::Uploader
            - add cpanminus and cpan-outdated
            - drop Perl::Critic
            - bump Log::Dispatchouli to 2
            - bump Moose to 1.19
            - replace Exception::Class with Throwable, Throwable::X, and
              Try::Tiny

20100611.000 2010-06-11 16:11:21 America/New_York
            - drop pip, Module::Install, YAML::XS
            - add perl 5.10.0
            - bump up some prereqs

20091203.000 2009-12-03
            - taskweaver now provides tests

20091129.000 2009-11-29 12:08:39 America/New_York
            - use TaskWeaver to build bundle

20090219     2009-02-19
            - dzil

1.003        2009-01-16
            - update my whims
            - add repo

1.002        2008-04-12
            - tell smokers to skip this
            - drop Rubric, which is huge and rarely needed
            - require 5.008 in Makefile.PL

1.001        2008-04-07
            - add version number
            - add changelog
            - add stupid tests

undef        2008-02-05
            - first release


lib/Task/BeLike/RJBS.pm
use 5.20.0;
use warnings;
package Task::BeLike::RJBS;
# ABSTRACT: be more like RJBS -- use the modules he likes!

=pkgroup Perl for Perl's Sake

=pkg perl 5.020 postfix dereference and much more

=pkg App::grindperl


=pkgroup Useful Command-Line Tools

=pkg App::Ack 1.82 a working --pager

=pkg App::Nopaste 1.004 Gist support with stricter paste validation

=pkg App::Whiff

App::Whiff provides C<whiff>, which replaces C<which>, because C<which> is
stupid.

=pkg App::Uni

This gets us the "uni" command, which lets you run "uni snowman" to copy and
paste the character and look funny on IRC.

=pkg File::Rename

This lets me rename a bunch of files by apply a C<s///> expression, or more.

=pkg Net::Dict

=pkg Unicode::Tussle

This is a whole bunch of extra utilities for poking through Unicode data.  It's
all cool stuff.


=pkgroup Tools for Working with the CPAN

=pkg perl 5.14.0 it has package NAME BLOCK

=pkg App::cpanminus 1.1002 it has --auto-cleanup

=pkg App::cpanoutdated 0.12 it won't install old dists

=pkg CPAN::Mini 0.563

=pkg Module::CPANfile

=pkg Module::Which

I use F<which_pm> to find the version and location of installed modules, even
if two versions are installed in different parts of C<@INC>.

=pkg Pod::Cpandoc

It's like C<perldoc>, but for stuff you haven't installed (yet?).

=pkg Test2::Harness


=pkgroup Tools for Building CPAN Distributions

=pkg Dist::Zilla 5 encoding!

=pkg Dist::Zilla::PluginBundle::RJBS 5 newest available

=pkg Perl::Tidy 20071205 supports 5.10

=pkg Pod::Weaver 4 encoding!



=pkgroup Application Frameworks

=pkg App::Cmd 0.308 it has App::Cmd::Setup bugfixes

App::Cmd also gets us Getopt::Long::Descriptive.



=pkgroup Email-Handling Libraries

=pkg Email::MIME 1.905 merged in Creator and Modifier modules; bug fixes

=pkg Email::Sender

=pkg Email::Sender::Transport::SQLite



=pkgroup Other Libraries I Use

=pkg Archive::Tar::Wrapper

Largely to suppress "it could be faster!" warnings from Dist::Zilla.

=pkg Authen::OATH

I need this for TOTP.

=pkg Authen::SASL

=pkg Config::INI 0.011

=pkg Convert::Base32

I need this for TOTP.

=pkg DBD::SQLite

=pkg Data::GUID 0.044 requires a new enough Data::UUID to work around Debian

=pkg Data::Printer

=pkg DateTime 0.51 provides CLDR support with fewest known bugs

=pkg Devel::Confess

=pkg Devel::Cover

=pkg Devel::NYTProf

=pkg HTML::Element 3.22 has proper XML escaping

=pkg IO::Async

=pkg JSON::MaybeXS

=pkg List::AllUtils

=pkg Log::Dispatchouli 2.000 it has Log::Dispatchouli::Global

=pkg Moose 1.19 it has assert_coerce

=pkg MooseX::StrictConstructor

All constructors must be strict!

=pkg namespace::autoclean

=pkg Net::Async::HTTP

=pkg PPI 1.212 fixes parsing of package names with leading-digit parts

=pkg Params::Util 0.38 has fixes to _IDENTIFIER and _CLASS

=pkg Plack

=pkg Scalar::Util 1.18

=pkg Sub::Exporter 0.980 INIT collector; bug fixes

=pkg Term::ReadLine::Gnu 1 improves the debugger

=pkg Text::Markdown 1.0.24 has trust_list_start

=pkg Throwable 0.102080 StackTrace::Auto factored out

=pkg Throwable::X

=pkg Time::Duration

=pkg Time::Duration::Parse

=pkg TOML::Parser

=pkg Try::Tiny 0.007 exception passed to C<finally>



=pkgroup Sanity-Check

These are just here to make sure other things work properly.

=pkg Mozilla::CA

=pkg LWP::Protocol::https

=pkg IO::Socket::SSL

=pkg Config::GitLike

=cut

1;


Makefile.PL
# You don't need to smoke my bundle, but thanks! -- rjbs, 2008-04-12
exit 0 if $ENV{AUTOMATED_TESTING};

BEGIN { require 5.008005; } # why would I use anything less?
use strict;
use warnings;

use inc::Module::Install;

name        ('Task-BeLike-RJBS');
all_from    ('lib/Task/BeLike/RJBS.pm');
extra_tests;

  requires('App::Ack',                       '1.76'  ); # 5.10 fix
                                                        # also gets File::Next
  requires('App::Cmd',                       '0.010' ); # App::Cmd::Simple
                                                        # Getopt::Long::Desc
  requires('App::Nopaste',                   '0.08'  ); # gist with token auth
  requires('App::Whiff',                     undef   );
  requires('CPAN',                           '1.87'  ); # homedir, sudo
  requires('CPAN::Mini',                     '0.563' ); # bugfixes
  requires('CPAN::Reporter',                 undef   );
  requires('CPAN::Uploader',                 undef   );
  requires('Config::INI',                    '0.011' ); # bugfixes
  requires('DBD::SQLite',                    undef   );
  requires('Data::GUID',                     '0.043' ); # debian--
  requires('DateTime',                       '0.41'  );
  requires('Dist::Zilla',                    '1.007' ); # metaresources
  requires('Dist::Zilla::Plugin::PodPurler', undef   );
  requires('Email::Filter',                  '1.02'  ); # bugfixes
  requires('Email::MIME',                    '1.861' ); # ensures Simple 2
  requires('Exception::Class',               '1.23'  ); # ->caught()
  requires('HTML::Element',                  '3.22'  ); # XML escaping
  requires('List::MoreUtils',                '0.21'  ); # bugfixes, part()
  requires('Log::Dispatchouli',              undef   );
  requires('Mixin::ExtraFields',             '0.005' ); # undef for id
  requires('Mixin::ExtraFields::Param',      '0.011' );
  requires('Module::CoreList',               '2.13'  ); # 5.10
  requires('Module::Install',                '0.74'  ); # bugfixes
  requires('Moose',                          '0.74'  ); # duck type
  requires('MooseX::POE',                    '0.200' ); # compat
  requires('PPI',                            '1.201' );
  requires('Params::Util',                   '0.23'  ); # _INVOCANT fixes
  requires('Perl::Critic',                   '1.080' );
  requires('Perl::Critic::Lax',              '0.007' );
  requires('Perl::Tidy'                      20071205); # 5.10
  requires('Pod::Eventual',                  '0.005' ); # bugfixes; Simple
  requires('Scalar::Util',                   '1.18'  ); # bugfixes
  requires('Sub::Exporter',                  '0.978' ); # installer/generator
  requires('Text::Markdown',                 '1.0.24'); # trust_list_start
  requires('YAML::XS',                       undef   );
  requires('pip',                            '0.12'  ); # pip http://...

if ($^O eq 'darwin') {
  # Mac Only
  requires('App::Addex',                     '0.013' );
  requires('App::Addex::AddressBook::Apple', '0.006' );
  requires('Mac::Glue',                      '1.30'  );
}

repository('http://github.com/rjbs/task-belike-rjbs');
auto_manifest;

WriteAll;
